python/tvm/relay/op/strategy/x86.py

```diff
@@ -507,10 +507,26 @@ def matmul_strategy_cpu(attrs, inputs, out_type, target):
     return strategy
 
 
+def is_any(shape):
+    return any([isinstance(x, tir.Any) or isinstance(x, tir.SizeVar) for x in shape])
+
+
 @dense_strategy.register("cpu")
 def dense_strategy_cpu(attrs, inputs, out_type, target):
     """dense x86 strategy"""
+
     strategy = _op.OpStrategy()
+    # For dynamic shapes we use a hand written kernel. Right now it only
+    # supports matrix-vector multiplication.
+    if is_any(inputs[0].shape) or is_any(inputs[1].shape):
```

Comment:
should we check the size of `inputs[0]` to make sure its a vector before dispatching?

Reply:
Added a check in the implementation.